x/swaprouter/creator.go

```diff
@@ -64,7 +67,10 @@ func (k Keeper) CreatePool(ctx sdk.Context, msg types.CreatePoolMsg) (uint64, er
 		return 0, err
 	}
 
-	k.poolCreationListeners.AfterPoolCreated(ctx, sender, pool.GetId())
+	// TODO: Ensure this is correct decision. Without this hack, we try to make gauges for the CL pool which isn't needed
+	if poolType != types.Concentrated {
+		k.poolCreationListeners.AfterPoolCreated(ctx, sender, pool.GetId())
+	}
```

Comment:
Let's create an issue for this and link it here directly. We should create a separate hook called `AfterCFMMPoolCreated`, call it in `gamm`'s `InitializePool`. The gauges should be created upon that new hook being called, not the `AfterPoolCreated`.